test/parser_spec.hs

```diff
@@ -22,6 +22,19 @@ import Test.QuickCheck.Monadic
 
 import Debug.Trace (trace)
 
+splitSizeAndReduce_ :: Int -> Gen (Int, Int)
```

Comment:
what does this do? :scream:

Reply:
it's for resizing of recursive Arbitrary instances 🙀  

So that it can generate random values of various types for property tests 😱

Reply:
without combinatorial explosion 😱😱😱